### ISSUE TITLE
Querying Via Relations Doesnt Seem to Work

### ISSUE THREAD

**Issue by donnfelker** [excid3/prefixed_ids#56]
Not sure if this is intended behavior or not, found via working with [JSP](https://jumpstartrails.com/). 

```ruby
class Account < ApplicationRecord
  has_prefix_id :acct
  has_many :customers, dependent: :destroy
end 

class Customer < ApplicationRecord
  has_prefix_id :cust
end
```

This doesn't seem to work in a controller: 
```ruby
current_account.customers.find(params[:id]) 
```

This returns this in the server logs when debugging into it: 
```
 current_account.customers.find(params[:id])
  Customer Load (0.3ms)  SELECT "customers".* FROM "customers" WHERE "customers"."account_id" = $1 AND "customers"."id" = $2 LIMIT $3  [["account_id", 1], ["id", nil], ["LIMIT", 1]]
  ↳ (rdbg)/path/to/app/controllers/customers_controller.rb:1:in `set_customer'
eval error: Couldn't find Customer with 'id'=cust_AZo14zMKyXJPah7YDnJbw6O3 [WHERE "customers"."account_id" = $1]
```

Is this intended behavior?

Please note, this does work though: 
```ruby
current_account.customers.find_by_prefix_id(params[:id])
```

**Update**
If I change the account model to this:
`has_prefix_id :acct, override_param: false`, the issue seems to go away. 

The problem manifests itself in JSP when the account has a prefix id and to_param is overriden, from what I can tell. Still doing more investigating an only fails on `rails test` for some of the tests. Will do more research and update this.

**Comment by excid3** [comment]
Hmm, will take a look. I believe that should work.

**Comment by excid3** [comment]
Just checked this in the `prefixed_ids` test suite which should match the Jumpstart Pro scenario and it worked fine. 🤔 

```ruby
User.first.accounts.find("acct_Egm1bKrB7DP60YLhXnmR4lnday8JX039")
=>
#<Account:0x00000001086f0780
 id: 113629430,
 user_id: 113629430,
 created_at: Tue, 30 Jan 2024 15:40:45.554993000 UTC +00:00,
 updated_at: Tue, 30 Jan 2024 15:40:45.554993000 UTC +00:00>
```

**Comment by excid3** [comment]
I'm not able to reproduce this in Jumpstart Pro either. 

```ruby
class Account < ApplicationRecord
  has_prefix_id :acct
  has_many :users
  
class User < ApplicationRecord
  has_prefix_id :user
```

**Comment by donnfelker** [comment]
Ok, looks like ID10T/PEBCAK error for the first part. So, this works as espected. I did something stupid. 🤦 

Apologies for wasting your time, Chris. 😬 

However, I was able to reproduce an issue on JSP. Add this to the `Account.rb` model: 
`has_prefix_id :acct` 

Then run `rails test` and a bunch of tests break. I'm going to close this and open an issue there as it more appropriate there, not here.
